src/Groups/matrices/MatGrp.jl

```diff
@@ -564,6 +573,17 @@ function order(::Type{T}, G::MatrixGroup) where T <: IntegerUnion
    return T(res)::T
 end
 
+# Use the reduction to a finite field as a finiteness test.
+function is_finite(G::MatrixGroup{T}) where {T <: Union{AbsSimpleNumFieldElem, QQFieldElem}}
+  try
+    compute_order(G)
+  catch e
+    return false
+  end
+  return true
```

Comment:
It would indeed be nicer to avoid the try/catch here, since it is unspecific: if some other error is thrown, then we won't recognize it and treat it as meaning "the group is infinite" even though it might be finite and the exception was due to a bug...

So perhaps `_isomorphic_group_over_finite_field` could be changed to return e.g. `nothing` to indicate the group is infinite, and then we roll with it? I.e. we could change `isomorphic_group_over_finite_field` to detect this return value; if so, it could immediately mark the group `G` as infinite. Otherwise, it stores the morphism as before. This way we basically get no overhead.

Alternatively we could change `_isomorphic_group_over_finite_field` to thrown an `InfiniteOrderError` (actually that might be a good idea to do anyway).